test/spoof/spoof.go

```diff
@@ -228,6 +247,11 @@ func (sc *SpoofingClient) Poll(req *http.Request, inState ResponseChecker) (*Res
 			}
 			if isConnectionReset(err) {
 				sc.Logf("Retrying %s for connection reset: %v", req.URL, err)
+				return false, nil
+			}
+			if retryOnAllErrors {
```

Comment:
I think we'd have significantly greater flexibility here if we offered a callback for evaluating whether to retry and the default value was `isConnectionReset` (which we'd want to make public to enable composition).

Not only would this let us plug in an option that allows retrying all errors, but also lets us customize the logging we do on such events, which feels like it matches your goal of "tracking downtime".